pkgs/development/libraries/mesa/default.nix

```diff
@@ -133,6 +133,29 @@ self = stdenv.mkDerivation {
 
     ./opencl.patch
     ./disk_cache-include-dri-driver-path-in-cache-key.patch
+  ] ++ lib.optionals stdenv.isDarwin [
```

Comment:
Note that these patches are not in the mesa main branch yet. I am hopeful, but ultimately we might have to disable darwin again if the patches stop applying and haven't been rebased/upstreamed and nobody puts in the effort to do that.